src/utils/removeInternalProperties.ts

```diff
@@ -28,7 +28,10 @@ export function removeInternalProperties<
         }
 
         // Remove internal properties of a "manyOf" relation.
-        if (Array.isArray(value)) {
+        if (
+          Array.isArray(value) &&
+          value.some((entity: any) => typeof entity === 'object')
```

Comment:
Would it be better to use `!values.every(isPrimitive)` here? It's not entirely clear to me what all of the possible values in this array could be, so wasn't sure how to best check this.

Reply:
Actually, this check was dropped entirely in https://github.com/mswjs/data/commit/369a816b7eccf5025a7fb8356b18db268cd5f38e and we now allow anything that's not an internal entity, one or manyOf relation when removing internal properties. This should cover the arrays as well.

Reply:
I've rebased your feature branch so this comment (and the change) is not applicable anymore.